cmd/kubeadm/app/cmd/init.go

```diff
@@ -87,6 +87,12 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 			i, err := NewInit(cfgPath, internalcfg, skipPreFlight, skipTokenPrint)
 			kubeadmutil.CheckErr(err)
 			kubeadmutil.CheckErr(i.Validate(cmd))
+
+			// TODO: remove this warning in 1.9
+			if !cmd.PersistentFlags().Lookup("token-ttl").Changed {
```

Comment:
Yeah, I think you're right. [The docs](https://godoc.org/github.com/spf13/cobra#Command.Flags) agree as well.